File: C/QE-Roots.c
#include <stdio.h>
#include <math.h>
#include "./delta.hrm"

int main()
{
  float a, b, c, d, x1, x2;
  printf("Insert Coeffs, space seperated: ");
  scanf("%f %f %f", &a, &b, &c);
  d = delta(a, b, c);
  if(d >= 0)
  {
    x1 = (-b - sqrt(d)) / (2 * a);
    x2 = (-b + sqrt(d)) / (2 * a);
    printf("roots = %f, %f\n", x1, x2);
  }
  else
  {
    printf("there is no real root\n");
  }

  return 0;
}

File: Python/QE-Roots.py
from math import sqrt

# ask user for coefficients
C = [input('insert coeff ' + str(i+1) + ' : ') for i in range(3)]

# Calculating delta
delta = C[1]**2 - 4 * C[0] * C[2]

_2a = 2 * C[0]

if delta >= 0:
    x1 = (-C[1] - sqrt(delta)) / _2a
    x2 = (-C[1] + sqrt(delta)) / _2a
else:
    x1 = str(-C[1] / _2a) + " + i * " + str(sqrt(-delta) / _2a)
    x2 = str(-C[1] / _2a) + " - i * " + str(sqrt(-delta) / _2a)

print(x1, x2)
